docs/features/shared-volumes-with-NFS.md

```diff
@@ -0,0 +1,138 @@
+
+### Update plugin configuration
+The plugin needs to be provided the details of the exported diretories and the servers from which to mount them. The details are entered into the plugin configuration file at /etc/docker-volume-vsphere.conf in the format as detailed later.
+
+### Restart plugin
+The plugin needs to be restarted to load the updated configuration.
+
+### Using shared volumes
+Docker volume commands must be used as shown below to use volumes on remote directories. Similar to how local VMDK based volumes are qualified with a datastore name, shared volumes must also be qualified with the label (see remote directory configuration below) that identifies the remote dir. to use to locate and refer a volume.
+1. docker volume create - 
+   1. _docker volume create --driver=vsphere --name=remote-vol -o type=file_ - uses the default entry in the list of remote dirs to locate the volume
+   1. _docker volume create --driver=vsphere --name=remote-vol@remote-dir-label_ - creates the volume in the location identified by the label in the remote dirs list
+1. docker volume list - No change
+1. docker volume inspect - the volume must be qualified by the remote dir label as below,
+   1. _docker volume inspect vol-name@remote-dir-label_
+1. docker run - the volume must be qualified by the remote dir label as below,
+   1. _docker run -it --volume-driver=vsphere -v remote-vol@remote-dir-label:/vol1 --name ub ubuntu_
+1. docker volume rm - the volume must be qualified by the remote dir label as below,
+   1. _dovker volume rm vol-name@remote-dir-label_
+
+
+# Extending the vSphere driver to support shared volumes
+
+The vSphere plugin driver supports an extendable framework that allows adding support for volumes backed by specific storage types. Presently the driver supports VMDK backed volumes on storage available to the local ESX host. A new module to support file backed volumes (on local or NFS mounted directories) is added to the existing framework that supports all of the docker volume functions as listed above.
+
+The new file backed volumes will support NFS v3/v4.0 protocols to mount and use remote directories to host docker volumes. Docker volumes will be supported as folders on the mounted directories and exported to containers in the same way that VMDK volumes are made available to containers.
+
+# Plugin configuration for remote directories
+
+The plugin configuration defined in /etc/docker-volume-vsphere.conf must be modified to include the details of the remote directories to use to locate volumes. The format describing a remote directory is as shown below,
+
+## /etc/docker-volume-vsphere.conf
+{
+"MaxLogAgeDays": 28,
+"MaxLogSizeMb": 100,
+"LogPath": "/var/log/docker-volume-vsphere.log",
+"LogLevel": "debug",
+_“RemoteDirs”: {_
```

Comment:
Yes, the only way is via the config file, I can add this via command line once the changes are submitted. Even authentication for nfs4 isn't included in the first set of the changes.